### PR TITLE
Session5

### PR DESCRIPTION
**Pull request by akosighie**

### CODE REVIEW

**Comment by kevinatgithub** [comment]
Can you please include me as reviewer, thanks

File: DotNetCoreMasters/Services/Implementation/ItemService.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+            IMapper mapper = configMapper.CreateMapper();
+            var mappedItemDTOEnumerable = mapper.Map(itemListRepo, itemListDTO);
+
+            return mappedItemDTOEnumerable;
+        }
+
+        public IEnumerable<ItemDTO> GetAllByFilter(ItemByFilterDTO filters)
+        {
+            throw new NotImplementedException();
```
Comment (jegtugado) [inline DotNetCoreMasters/Services/Implementation/ItemService.cs]: needs implementation

File: DotNetCoreMasters/Repositories/Implementation/ItemRepository.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Repositories.Implementation
+{
+    public class ItemRepository : IItemRepository
+    {
+
+        public ItemRepository()
```
Comment (jegtugado) [inline DotNetCoreMasters/Repositories/Implementation/ItemRepository.cs]: inject data context and use it to implement the methods

File: DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs [hunk 3]
```diff
@@ -8,16 +12,64 @@
...
+        {
+            var listItems = _itemService.GetAll().ToList();
+
+            return Ok(listItems);
+        }
+
+        [HttpGet("/items/{itemId}")]
+        public IActionResult Get(int itemId)
+        {
+
```
Comment (jegtugado) [inline DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs:34]: use item service

File: DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs [hunk 4]
```diff
@@ -8,16 +12,64 @@
...
+            var model = itemCreateModel;
+
+            return Ok();
+        }
+
+        [HttpDelete("/items/{itemId}")]
+        public IActionResult Put(int itemId)
+        {
+            var newId = itemId;
+
```
Comment (jegtugado) [inline DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs:80]: use item service

File: DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs [hunk 5]
```diff
@@ -8,16 +12,64 @@
...
+            var model = itemCreateModel;
+
+            return Ok();
+        }
+
+        [HttpPut("/items/{itemId}")]
+        public IActionResult Put(int itemId, [FromBody] ItemCreateBindingModel itemCreateModel)
+        {
+            var newId = itemId;
+            var model = itemCreateModel;
```
Comment (jegtugado) [inline DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs]: use item service

File: DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs [hunk 6]
```diff
@@ -8,16 +12,64 @@
...
 
-            return Ok(listOfStrings);
+            return Ok(filter);
         }
+
+        [HttpPost]
+        public IActionResult Post([FromBody] ItemCreateBindingModel itemCreateModel)
+        {
+            var model = itemCreateModel;
+
```
Comment (jegtugado) [inline DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs:53]: use item service

**Review by jegtugado — CHANGES_REQUESTED** [review]
use the item service in itemscontroller. implement each method in your service and repository.

File: DotNetCoreMasters/Services/DI/ItemCollectionService.cs [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+using System.Text;
+
+namespace Services.DI
+{
+    public static class ItemCollectionService
+    {
+        public static IServiceCollection AddItemService(this IServiceCollection services) //extend IServiceCollection 
+        {
+            services.AddScoped<IItemService, ItemService>();
+            services.AddScoped<IItemRepository, ItemRepository>();
```
Comment (jegtugado) [inline DotNetCoreMasters/Services/DI/ItemCollectionService.cs:18]: Where is the registration for the DataContext?

File: DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs [hunk 8]
```diff
@@ -8,16 +12,64 @@
...
+        [HttpGet]
+        public IActionResult GetAll()
+        {
+            var listItems = _itemService.GetAll().ToList();
+
+            return Ok(listItems);
+        }
+
+        [HttpGet]
```
Comment (jegtugado) [inline DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs]: where is the route parameter?

File: DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs [hunk 9]
```diff
@@ -8,16 +12,64 @@
...
+        public IActionResult Get(int itemId)
+        {
+
+            return Ok(itemId);
+        }
+
+
+        [HttpGet]
```
Comment (jegtugado) [inline DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs]: where is the routing? refer to the homework

File: DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs [hunk 10]
```diff
@@ -8,16 +12,64 @@
...
+        public IActionResult GetByFilters([FromQuery] Dictionary<string, string> filters)
         {
-            var res = new ItemService();
-            var listOfStrings = res.GetAll(userId);
+            var filter = filters;
 
-            return Ok(listOfStrings);
+            return Ok(filter);
```
Comment (jegtugado) [inline DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs]: where is the call to item service?

File: DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs [hunk 11]
```diff
@@ -8,16 +12,64 @@
...
+        [HttpPut]
+        public IActionResult Put(int itemId, [FromBody] ItemCreateBindingModel itemCreateModel)
+        {
+            var newId = itemId;
+            var model = itemCreateModel;
+
+            return Ok();
+        }
+
+        [HttpDelete]
```
Comment (jegtugado) [inline DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs]: route parameter? action name should be Delete

File: DotNetCoreMasters/Services/Implementation/ItemService.cs (hunk 1 above)
Comment (jegtugado) [inline DotNetCoreMasters/Services/Implementation/ItemService.cs]: implement

File: DotNetCoreMasters/Repositories/Implementation/ItemRepository.cs [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+        {
+            var item = ListOfItems().Where(i => i.ItemId == id).SingleOrDefault();
+            ListOfItems().Remove(item);
+        }
+
+        public void Save(Item item)
+        {
+            var isExist = ListOfItems().Where(i => i.ItemId == item.ItemId).Any();
+
+            if (!isExist)
```
Comment (jegtugado) [inline DotNetCoreMasters/Repositories/Implementation/ItemRepository.cs:39]: handle both add and update scenarios

**Review by jegtugado — CHANGES_REQUESTED** [review]
- no route parameters
- wrong action method name
- no item service calls
- missing implementations in item service
- missing implementation in item repository

File: DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs [hunk 13]
```diff
@@ -16,10 +16,12 @@ namespace DotNetCoreMasters.Controllers
     public class ItemsController : ControllerBase
     {
         private readonly IItemService _itemService;
+        private readonly ItemDTO _itemDTO;
 
-        public ItemsController(IItemService itemService)
+        public ItemsController(IItemService itemService, ItemDTO itemDTO, ItemByFilterDTO itemDTOFilter)
```
Comment (jegtugado) [inline DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs]: why add these parameters to the constructor?

File: DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs (hunk 11 above)
Comment (jegtugado) [reply DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs]: still no route parameter on the Route attribute. make sure the itemId is same with other routes

File: DotNetCoreMasters/DotNetCoreMasters/Startup.cs [hunk 14]
```diff
@@ -19,6 +20,8 @@ public void ConfigureServices(IServiceCollection services)
         {
             services.AddControllers();
             services.AddItemService();
```
Comment (jegtugado) [inline DotNetCoreMasters/DotNetCoreMasters/Startup.cs:22]: what is this for?

File: DotNetCoreMasters/Repositories/Implementation/ItemRepository.cs [hunk 15]
```diff
@@ -9,49 +10,47 @@ namespace Repositories.Implementation
 {
     public class ItemRepository : IItemRepository
     {
-
-        public ItemRepository()
+        private IItemContext _itemContext;
+        public ItemRepository(IItemContext itemContext)
```
Comment (jegtugado) [inline DotNetCoreMasters/Repositories/Implementation/ItemRepository.cs]: where did IItemContext come from? It should be DataContext

**Review by jegtugado — CHANGES_REQUESTED** [review]
still needs lots of correction. comment on the PR when everything is solved as to not prematurely check whenever there is a change

File: DotNetCoreMasters/DotNetCoreMasters/Startup.cs (hunk 14 above)
Comment (akosighie) [reply DotNetCoreMasters/DotNetCoreMasters/Startup.cs:22]: Register DI for service

Comment (jegtugado) [reply DotNetCoreMasters/DotNetCoreMasters/Startup.cs:22]: i mean this line 'services.AddScoped<ItemDTO>();'

Comment (jegtugado) [reply DotNetCoreMasters/DotNetCoreMasters/Startup.cs:22]: for ItemDTO

Comment (akosighie) [reply DotNetCoreMasters/DotNetCoreMasters/Startup.cs:22]: updated no services.AddScoped();

File: DotNetCoreMasters/Repositories/Implementation/ItemRepository.cs (hunk 15 above)
Comment (akosighie) [reply DotNetCoreMasters/Repositories/Implementation/ItemRepository.cs]: done renaming to DataContext

**Review by jegtugado — APPROVED** [review]

**Review by jegtugado — APPROVED** [review]

**Pull request merged**